crates/red_knot_test/src/lib.rs

```diff
@@ -30,8 +32,13 @@ pub fn run(path: &Path, long_title: &str, short_title: &str) {
 
     let mut db = db::Db::setup(SystemPathBuf::from("/src"));
 
+    let filter = std::env::var(MDTEST_FILE_FILTER).ok();
     let mut any_failures = false;
     for test in suite.tests() {
+        if filter.as_ref().is_some_and(|f| !test.name().contains(f)) {
```

Comment:
just doing a simple case-sensitive substring check. this could also be changed to do a glob or regex, though then we have to think about escaping if we want to print the expected filter for a given test. shouldn't be too big of a deal, so happy to implement another solution if we want to